auth/collection_access_test.go

```diff
@@ -0,0 +1,201 @@
+// Copyright 2022-Present Couchbase, Inc.
+//
+// Use of this software is governed by the Business Source License included
+// in the file licenses/BSL-Couchbase.txt.  As of the Change Date specified
+// in that file, in accordance with the Business Source License, use of this
+// software will be governed by the Apache License, Version 2.0, included in
+// the file licenses/APL2.txt.
+
+package auth
+
+import (
+	"bytes"
+	"log"
+	"testing"
+
+	"github.com/couchbase/sync_gateway/base"
+	ch "github.com/couchbase/sync_gateway/channels"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func canSeeAllCollectionChannels(scope, collection string, princ Principal, channels base.Set) bool {
```

Comment:
nice to have: pass t in here, and assert on type, so you know which channel you can't see is